File: mesh-delivery/android/ExoPlayerMeshDeliveryJava/app/src/main/java/io/streamroot/lumen/delivery/client/samples/mesh_delivery/exoplayer/MainActivity.java
package io.streamroot.lumen.delivery.client.samples.mesh_delivery.exoplayer;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import io.streamroot.lumen.delivery.client.samples.mesh_delivery.exoplayer.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        final ActivityMainBinding binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        binding.launchButton.setOnClickListener(v -> {
            final String stream = binding.streamEditText.getText().toString();
            if (stream.isEmpty()) return;
            final Intent i = PlayerActivity.makeIntent(MainActivity.this, new PlayerActivity.PlayerActivityArgs(
                            binding.dcKeyET.getText().toString(),
                            stream,
                            binding.meshPropET.getText().toString()
                    )).addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT)
                    .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            startActivity(i);
        });
    }
}

File: mesh-delivery/android/ExoPlayerMeshDeliveryPlugin/app/src/main/java/io/streamroot/lumen/delivery/client/samples/mesh_delivery/exoplayer/MainActivity.kt
package io.streamroot.lumen.delivery.client.samples.mesh_delivery.exoplayer

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import io.streamroot.lumen.delivery.client.samples.mesh_delivery.exoplayer.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        ActivityMainBinding.inflate(layoutInflater).apply {
            this.launchButton.setOnClickListener {
                if (streamEditText.text.isBlank()) return@setOnClickListener;
                PlayerActivity.makeIntent(
                    this@MainActivity, PlayerActivity.PlayerActivityArgs(
                        dcKeyET.text.toString(),
                        streamEditText.text.toString(),
                        meshPropET.text.toString()
                    )
                ).apply {
                    addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT)
                    addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
                }.run {
                    startActivity(this)
                }
            }

            setContentView(this.root)
        }
    }
}